### ISSUE TITLE
Reasonable balances for trading

### ISSUE THREAD

**Issue by xeroquark** [notadamking/Bitcoin-Trader-RL#38]
Currently, agents buy and sell using unrealistic USD and BTC balances. 
For instance, during a debugging session I saw one agent executing this kind of trades:

>Buying BTC
balance USD: 0.004812330813060336 btc held: 2.5835011089138864
Net worth: 17707.373082848768
=============
>Buying BTC
balance USD: 0.0032082205420402242 btc held: 2.5835013414545522
Net worth: 17777.0501037559

It doesn't look like there is a logical cap on the USD and BTC that can be used in a trade. In real life one wouldn't try to (and couldn't) buy BTC with $0.0016.

**Comment by silentrob** [comment]
Good catch. I think there is some room for improvement on picking the position size.

**Comment by arron202** [comment]
Yes, I also meet a similar issue:
Bought: -9.094947017729282e-13
also during optimization on trial have the results seems overflow:
Best trial: 1014384906207232.0

We may have a least trade unit like a hand in stock, if our Net worth is less than that, it should stop trading.

**Comment by ktattan** [comment]
Is the bet size determined by the [amount variable](https://github.com/notadamking/Bitcoin-Trader-RL/blob/master/env/BitcoinTradingEnv.py#L104) or the [12 discrete action spaces](https://github.com/notadamking/Bitcoin-Trader-RL/blob/master/env/BitcoinTradingEnv.py#L59)? I am having a hard time understanding how the bet size is calculated, and why 12 was chosen as the size of the discrete action space.

**Comment by notadamking** [comment]
Great catch @xeroquark. I will gladly accept a PR fixing this issue.

@ktattan 12 discrete options allows the agent to buy using 25% increments of its balance, sell in 25% increments of its currently held BTC, or hold. There is definitely room for improvement in this action space strategy, perhaps a simple buy all/sell all/hold would perform better.
